src/liboslexec/runtimeoptimize.cpp

```diff
@@ -2121,6 +2122,26 @@ RuntimeOptimizer::optimize_ops (int beginop, int endop,
         // Nothing below here to do for no-ops, take early out.
         if (op->opname() == u_nop)
             continue;
+        // If an exit or return was reached, kill the remaining instructions
+        if (optimize() && (op->opname() == u_return || op->opname() == u_exit)) {
```

Comment:
I'm not sure this is correct. For exit, yes, but for `return`, it would be wrong to eliminate all the rest of the ops in the shader. You would want to clear at most the rest of the ops in the function, right?

Reply:
Yes, perhaps the comment is wrong.
`optimize_ops`  will be called recursively for functions, with endop pointing to the function end.